lib/strings.nix

```diff
@@ -157,6 +157,29 @@ rec {
   */
   replicate = n: s: concatStrings (lib.lists.replicate n s);
 
+  /*
+    Remove leading and trailing whitespace from a string.
+
+    Whitespace is defined as any of the following characters:
+      " ", "\t" "\r" "\n"
+
+    Type: trim :: string -> string
+
+    Example:
+      trim "   hello, world!   "
+      => "hello, world!"
+  */
+  trim = s:
```

Comment:
what's "chomp" vs "trim"?

Reply:
Basically,

```nix
chompBegin = s:
  trim s true false;
chompEnd = s:
  trim s false true;
chomp = s:
  trim s true true;
```

Reply:
We could have a configurable `trim'`:

```nix
trim = trim' {
  start = true;
  end = true;
};

trim' =
  {
    start ? false,
    end ? false,
    ... # forward compatible
  }:
  s:
  # implementation
```

With this approach idk whether or not having `trimStart` and `trimEnd` has any value, since the configurable version can be used.

Not sure if it's better to have `start` and `end` arguments default to true or false though?

Reply:
Nice idea!

This "general function plus typical usage" approach is typical around Nixpkgs. We have `getExe'` plus `getExe`, after all.

Putting `begin` and `end` as false by default is most natural, since they will be overridden anyway: if I want to trim the beginning but not the ending, it would be not so ergonomical to set both explicitly.